src/Stormpath/DataStore/DefaultDataStore.php

```diff
@@ -121,6 +121,28 @@ public function getResource($href, $className, array $options = array())
             $this->addDataToCache($data, $queryString);
         }
 
+        if (isset($options['propertyId']))
```

Comment:
This class shouldn't be resolving the class name, so I would implement some type of `ClassNameResolver` (probably based on the provided className and the options array).